packages/plugin-legacy/index.js

```diff
@@ -170,7 +174,7 @@ function viteLegacyPlugin(options = {}) {
       }
       config = _config
 
-      if (!genLegacy) {
+      if (!genLegacy || config.build.ssr) {
```

Comment:
Why not do similar pattern as in `legacy-generate-polyfill-chunk`? Just set `config = _config` immediately and then only do the config.build.lib check if `!config.build.ssr`?

Reply:
Because it makes no sense to use `lib` mode in conjunction with either SSR or `plugin-legacy`, and we'd need to `return` somewhere anyway. In `legacy-generate-polyfill-chunk` `configResolved()` we don't need to `return` since there's no code following that and skipping the check is enough.